File: connect_socket/tcp_serverside.py
# import socket
#
# HOST = '127.0.0.1'
# PORT = 3020
#
# with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
#     s.bind((HOST, PORT))
#     s.listen()
#     conn, addr = s.accept()
#     with conn:
#         print('Connected by', addr)
#         while True:
#             data = conn.recv(1024)
#             if not data:
#                 break
#             conn.sendall(data)

import socket

def handle_data(data):
    ENQ = b'\x05'
    ACK = b'\x06'
    EOT = b'\x04'
    LF = b'\x0A'
    for byte in data:
        if byte == ENQ:
            print('RX ENQ')
            s.send(ACK)
        elif byte == EOT:
            print('RX EOT')
            s.send(ACK)
        elif byte == ACK:
            print('RX ACK')
            s.send(EOT)
        elif byte == LF:
            print('RX LF')
            s.send(ACK)

if __name__ == '__main__':
    HOST = '127.0.0.1'
    PORT = 3020

    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
        s.bind((HOST, PORT))
        s.listen()

        print(f'Server is listening on {HOST}:{PORT}')

        try:
            while True:
                conn, addr = s.accept()

                print(f'New connection from: {addr}')

                with conn:
                    print('Connected by', addr)
                    data = conn.recv(1024)
                    handle_data(data)
                    print(data)
        except Exception as e:
            print(e)
            s.close()

        except KeyboardInterrupt:
            s.close()

File: contoh_kode/tcp_serverside.js
'use strict';
const net = require('net');
const fs = require('fs');
var moment = require('moment');
//const server = net.createServer({port: 5833, host: '192.168.3.137'});

var STX = [0x02];
var ENQ = [0x05]; //ENQUIRY
var ACK = [0x06]; //ACKNOWLEDGE
var EOT = [0x04]; //END OF TEXT
var LF = [ 0x0A]; //LINE FEED
var ETX = [0x03];
var FS = [0x1C];

server.listen(6033, function () {
     console.log('server is listening');
 });

server.on('connection', socket => {
  console.log('new client arrived');
  socket.on("data", data => {
    console.log(data);
    for (var i = 0; i < data.length; i++){ 
         if (data[i] === 0x05) {
                console.log(data)
                socket.write(Buffer.from(String.fromCharCode(0x06)));
         }
         if (data[i] === 0x04) {
                console.log(data)
                socket.write(Buffer.from(String.fromCharCode(0x05)));
        }
        if (data[i] === 0x06) {
                console.log(data)
                socket.write(Buffer.from(String.fromCharCode(0x04)));
        }
        if (data[i] === 0x0A) {
                console.log(data)
                socket.write(Buffer.from(String.fromCharCode(0x06)));
        }
    }
    })
  server.on('close', () => {
      console.log('close, client disconnected')
  });

  server.on('end', () => {
      console.log('end')
  });

  server.on('error', () => {
      console.log('error, disconnected')
});
});
